desktop/libs/aws/src/aws/s3/s3fs.py

```diff
@@ -362,20 +362,19 @@ def rmtree(self, path, skipTrash=True):
       key = self._get_key(path, validate=False)
 
       if key.exists():
-        to_delete = [key]
         dir_keys = []
 
         if self.isdir(path):
-          dir_keys = key.bucket.list(prefix=path)
-          to_delete = itertools.chain(dir_keys, to_delete)
+          _, dir_key_name = s3.parse_uri(path)[:2]
+          dir_keys = key.bucket.list(prefix=dir_key_name)
```

Comment:
noob question: why not delete the bucket completely instead of deleting specific keys?

Reply:
There are scenarios where we don't want to delete full bucket consisting of all keys, instead some keys present in that specific bucket.